File: Domain/Models/Transactions/Transaction.cs
using Domain.Enums.Transactions;

namespace Domain.Models.Transactions;

public class Transaction
{
    /// <summary>
    /// The id
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// The title of the transaction
    /// </summary>
    public required string Title { get; set; }

    /// <summary>
    /// The amount of the transaction
    /// </summary>
    public double Amount { get; set; }

    /// <summary>
    /// The date and time of the transaction
    /// </summary>
    public DateTime Date { get; set; }

    /// <summary>
    /// The id of the account
    /// </summary>
    public Guid AccountId { get; set; }

    /// <summary>
    /// The <see cref="Payee"/> of this transaction
    /// </summary>
    public Payee? Payee { get; set; }
}

File: client-app/src/models/transaction.ts
import { Payee } from "./payee";

export interface Transaction {
    id: string;
    title: string;
    amount: number;
    date: string;
    accountId: string;

    payee: Payee | null;

    budgetItemId: string | undefined
}

export class Transaction implements Transaction {

    constructor(id: string, title: string, amount: number, date: string, accountId: string, payee: Payee, budgetItemId: string | undefined = undefined) {
        this.id = id;
        this.title = title;
        this.amount = amount;
        this.date = date;
        this.accountId = accountId;
        this.payee = payee;
        this.budgetItemId = budgetItemId;
    }
}